sct-commons/src/main/java/org/lfenergy/compas/sct/commons/ExtRefEditorService.java

```diff
@@ -399,4 +413,40 @@ private String computeDaiValue(AbstractLNAdapter<?> lnAdapter, TExtRef extRef, S
         }
     }
 
+    @Override
+    public void debindCompasFlowsAndExtRefsBasedOnVoltageLevel(SCL scd) {
+        LdeviceService ldeviceService = new LdeviceService();
+        scd.getSubstation()
+                .stream()
+                .flatMap(tSubstation -> tSubstation.getVoltageLevel().stream())
+                .filter(tVoltageLevel -> !"0".equals(tVoltageLevel.getName()))
+                .forEach(tVoltageLevel -> {
+                    String flowSource = voltageCodification.get(tVoltageLevel.getName());
+                    scd.getIED().stream()
+                            .flatMap(ldeviceService::getLdevices)
+                            .forEach(tlDevice -> {
```

Comment:
you can put tCompasFlow.isSetFlowSourceVoltageLevel() and isSetExtRefiedName in a filter condition, as it is used in both case : 
```java
PrivateUtils.getPrivateStream(tInputs.getPrivate(), TCompasFlow.class)
                                    .filter(TCompasFlow::isSetFlowSourceVoltageLevel)
                                    .filter(TCompasFlow::isSetExtRefiedName)
                                    .forEach(tCompasFlow -> {
```